fendermint/eth/api/src/conv/from_eth.rs

```diff
@@ -23,37 +24,72 @@ pub fn to_fvm_message(tx: TypedTransaction) -> JsonRpcResult<Message> {
+) -> Result<et::Transaction, JsonRpcError> {
+    match &tx {
+        TypedTransaction::Legacy(tx) => {
+            Ok(et::Transaction {
+                hash,
+                nonce: tx.nonce.unwrap_or_default(),
+                block_hash: None,
+                block_number: None,
+                transaction_index: None,
+                from: tx.from.unwrap_or_default(),
+                to: tx.to.clone().and_then(|to| to.as_address().cloned()),
+                value: tx.value.unwrap_or_default(),
+                gas: tx.gas.unwrap_or_default(),
+                // Strictly speaking a "Type 2" transaction should not need to set this, but we do because Blockscout
+                // has a database constraint that if a transaction is included in a block this can't be null.
+                gas_price: tx.gas_price,
+                input: tx.data.clone().unwrap_or_default(),
+                chain_id: tx.chain_id.map(|x| et::U256::from(x.as_u64())),
+                v: et::U64::from(sig.v),
+                r: sig.r,
+                s: sig.s,
+                transaction_type: Some(0u64.into()),
+                access_list: None,
+                other: Default::default(),
+                max_fee_per_gas: None,
+                max_priority_fee_per_gas: None,
+            })
+        }
+        TypedTransaction::Eip1559(tx) => {
+            Ok(et::Transaction {
+                hash,
+                nonce: tx.nonce.unwrap_or_default(),
+                block_hash: None,
+                block_number: None,
+                transaction_index: None,
+                from: tx.from.unwrap_or_default(),
+                to: tx.to.clone().and_then(|to| to.as_address().cloned()),
+                value: tx.value.unwrap_or_default(),
+                gas: tx.gas.unwrap_or_default(),
+                max_fee_per_gas: tx.max_fee_per_gas,
+                max_priority_fee_per_gas: tx.max_priority_fee_per_gas,
+                // Strictly speaking a "Type 2" transaction should not need to set this, but we do because Blockscout
+                // has a database constraint that if a transaction is included in a block this can't be null.
+                gas_price: Some(
+                    tx.max_fee_per_gas.unwrap_or_default()
+                        + tx.max_priority_fee_per_gas.unwrap_or_default(),
+                ),
+                input: tx.data.clone().unwrap_or_default(),
+                chain_id: tx.chain_id.map(|x| et::U256::from(x.as_u64())),
+                v: et::U64::from(sig.v),
+                r: sig.r,
+                s: sig.s,
+                transaction_type: Some(2u64.into()),
+                access_list: Some(tx.access_list.clone()),
+                other: Default::default(),
+            })
+        }
+        _ => error::error_with_revert(
+            ExitCode::USR_ILLEGAL_ARGUMENT,
+            "txn type not supported",
+            None::<Vec<u8>>,
         ),
```

Comment:
Why are we duplicating this logic here? `send_raw_transaction` is already producing this error?